vaadin-button-flow-parent/vaadin-button-flow-integration-tests/src/test/java/com/vaadin/flow/component/button/tests/ButtonIT.java

```diff
@@ -347,16 +326,23 @@ public void removeDisabled_buttonWorksNormally() {
     }
 
     @Test
-    public void disableOnClick_enableInSameRoundtrip_clientSideButtonIsEnabled() {
-        WebElement button = layout
-                .findElement(By.id("disable-on-click-re-enable-button"));
+    public void disableOnClick_enableInSameRoundTrip_clientSideButtonIsEnabled() {
+        var itemId = "disable-on-click-re-enable-button";
+        waitForElementPresent(By.id(itemId));
+        waitUntil(ExpectedConditions
+                .elementToBeClickable(findElement(By.id(itemId))), 2);
         for (int i = 0; i < 3; i++) {
-            Boolean disabled = (Boolean) executeScript(
-                    "arguments[0].click(); return arguments[0].disabled",
-                    button);
+            var button = findElement(By.id(itemId));
+            var disabled = (Boolean) getCommandExecutor().getDriver()
+                    .executeAsyncScript("""
```

Comment:
Could this be split into two executeScript to avoid `requestAnimationFrame` need?

Reply:
Just splitting it doesn't seem to work. I couldn't make it work in a simpler way either. Do you have a suggestion?

Reply:
Oh, right, requestAnimationFrame is needed not to wait for a server request but for the JS logic to apply the disabled attribute. Anyway, here is a slightly simpler version of the test that came up during my experiments:

```java
getCommandExecutor().disableWaitForVaadin();
ButtonElement button = $(ButtonElement.class)
        .id("disable-on-click-re-enable-button");
for (int i = 0; i < 3; i++) {
        button.click();
        getCommandExecutor().getDriver()
                .executeAsyncScript("requestAnimationFrame(arguments[0])");
        Assert.assertTrue(button.hasAttribute("disabled"));
        waitUntil(driver -> button.isEnabled());
}
getCommandExecutor().enableWaitForVaadin();
```

Reply:
Updated the tests to simplify this logic.